### PR TITLE
fix abi bug for 2.x 

### PR DESCRIPTION
**Pull request by lightszero**
fixed https://github.com/neo-project/neo-devpack-dotnet/issues/120

### CODE REVIEW

File: Neo.Compiler.MSIL/FuncExport.cs [hunk 1]
```diff
@@ -163,8 +163,9 @@ public static MyJson.JsonNode_Object Export(NeoModule module, byte[] script)
                         item.SetDictValue("type", ptype);
                     }
                 }
-                var rtype = ConvType(mm.returntype);
-                funcsign.SetDictValue("returntype", rtype);
+                //event do not have returntype in nep3
```
Comment (devhawk) [inline Neo.Compiler.MSIL/FuncExport.cs]: Should we be validating that event delegates are void return rather than simply not converting the return type?

Comment (lightszero) [reply Neo.Compiler.MSIL/FuncExport.cs]: that is no return type in nep3，so have returntype is a bug

Comment (devhawk) [reply Neo.Compiler.MSIL/FuncExport.cs]: right, so if the developer specifies an event with a return type other than void, that should be a compile failure. Is that checked elsewhere in NEON?

Comment (lightszero) [reply Neo.Compiler.MSIL/FuncExport.cs]: no,you are right,should be checked.let us make another issue for it.

Comment (devhawk) [reply Neo.Compiler.MSIL/FuncExport.cs]: filed #123 to track compile failure

File: Neo.Compiler.MSIL.UnitTests/TestClasses/Contract_appcall.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+using Neo.SmartContract.Framework;
+
+namespace Neo.Compiler.MSIL.TestClasses
+{
+    class Contract_syscall : SmartContract.Framework.SmartContract
```
Comment (shargon) [inline Neo.Compiler.MSIL.UnitTests/TestClasses/Contract_appcall.cs]: The name of the class and the file name are different

File: Neo.Compiler.MSIL.UnitTests/UnitTest_Appcall.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+namespace Neo.Compiler.MSIL
+{
+    [TestClass]
+    public class UnitTest_AppCall
+    {
+        [TestMethod]
+        public void Test_Appcall()
+        {
+            //var hash = UInt160.Parse("0102030405060708090A0102030405060708090A");
```
Comment (shargon) [inline Neo.Compiler.MSIL.UnitTests/UnitTest_Appcall.cs]: I prefer to remove this file. These changes are duplicated with https://github.com/neo-project/neo-devpack-dotnet/pull/121 ?

**Comment by lightszero** [comment]
I will fix unittest problem after
https://github.com/neo-project/neo-devpack-dotnet/pull/122#issuecomment-549934301

**Comment by lapetina9** [comment]
## Neo PR Test Report

**Repository:** neo-devpack-dotnet  
**Number:** 122  
**Title:** Fix ABI bug for 2.x     
**Change description:** Fixed #120

## Code version / branch

**Neo-devpack-dotnet:**  Branch_master-2.x_fixabi  


## Test Results

| Test Name | Network | OS | Test Result | Comments |
| ------- |:-------:| --:| -----------:| --------:|
| Initial UT | UT | Localhost | OK | 
| Additional unit test using delegate event with arguments | UT | Localhost | OK | |
| Additional unit test using action and delegate events with arguments | UT | Localhost | OK |  |
| Additional unit test using delegate event without arguments | UT| Localhost | OK |  |
| Additional unit test using inner class events | UT| Localhost | OK |  |

#### Additional info
The ABI is not deployed on Neo 2, so there is no need to deploy it on any network.


## **Test List**

**1. Initial UT**  
Run solution unit tests.

**2. Additional unit test to delegate event with arguments**  
  
Create Smart Contract with delegate event with arguments.

Expected result:
- Event data is written with arguments in abi file

**3. Additional unit test to action and delegate events with arguments**  

Create Smart Contract with action and delegate events with arguments.

Expected result:  
- Data from both events are written with arguments in abi file.  

**4. Additional unit test to delegate event without arguments**  

Create Smart Contract with delegate event without arguments.

Expected results:
- Event data is written in abi file.
- Arguments field is empty.

**5. Additional unit test to inner class events**  

Create Smart Contract with delegate event inside a inner class.

Expected results:
- Event data is written in abi file.

**Comment by shargon** [comment]
@lightszero conflicts

**Comment by lightszero** [comment]
how to fix travis error?

**Comment by shargon** [comment]
Wait for https://github.com/neo-project/neo-devpack-dotnet/pull/134

**Comment by shargon** [comment]
Ready to merge?

**Pull request merged**